models/clm/shell_scripts/cesm2_2/assimilate.csh

```diff
@@ -0,0 +1,387 @@
+#!/bin/csh
+#
+# DART software - Copyright UCAR. This open source software is provided
+# by UCAR, "as is", without charge, subject to all terms of use at
+# http://www.image.ucar.edu/DAReS/DART/DART_download
+#
+# This script performs an assimilation by directly reading and writing to
+# the CLM restart file. There is no post-processing step 'dart_to_clm',
+# consequently, snow DA is not supported in this framework.
+
```

Comment:
This comment is a bit confusing, because the main purpose of this update is to support snow DA.  I suppose it means if this script is run alone -- without support of clm_to_dart and dart_to_clm etc. snow DA is not possible?

Reply:
dart_to_clm  is the logical choice to implement the snow_update() mechanism, but it also replaces the CLM indeterminate values in the unused snow layers (and maybe elsewhere).  So right now - updating the prognostic variables based on increments to SWE is not possible. 
Updating the prognostic snow variables based on their ensemble correlation with (any) close observation is what happens currently (and may be the source of the 'soil balance errors').